### DISCUSSION TITLE
Picking The Right Codec

### DISCUSSION THREAD

**Discussion by stillusingmp3** [enzo1982/freac#540] [Audio Format Discussion]
Just curious which lossy codec would be best for encoding pipe organ music.

**Comment by Lee-Carre** [comment]
Why lossy instead of lossless?

Depends on several things;
* is this music you've acquired from elsewhere (eg CD), and simply want to put on something like a portable player (in which I include nano-computers (‘smart-phones’) for the purposes here)?
* is this music which you've recorded yourself, of an organist playing a real organ (eg in a church, or similar)? — Especially in this case, I would strongly recommend using lossless (and a libre lossless codec, at that) for the original/master (which other formats can be derived)

Separately from that (and somewhat assuming a non-recording scenario); if you're not constrained by your playback device, and may choose any codec, then I would suggest the following:
* libre codecs over proprietary ones
* modern codecs over older ones
* assuming that you're doing things generally sensibly (no weird edge-cases), then lossy encoding/compression will benefit much more from clean (ie as noise-free as possible; especially if recording, then ideally the recorder fed from a line-out signal, rather than microphones, unless you want all the reverb you'll get in a church, but it'll be more difficult to efficiently encode (read: larger files or lower-fidelity)) recordings, and this will have a greater effect than choice of codec
* other considerations include; are you storage-space constrained (need small output files), or not?

Short of knowing more (but please do elaborate), then to give a single, simple answer (for lossy compression, at least), I'd recommend Opus.

**Reply by stillusingmp3** [reply]
I ripped the music from CD to wav files first then decide later what to convert them to.
The music will be transferred over to a portable player ( Astel & Kern SP3000 ) using the Focal Utopia headphones.
So Going by your reply i have decided to just keep them as wav files since there's plenty of storage on the device.
Thank you so much for answering my query on the subject.

**Reply by Lee-Carre** [reply]
> music from CD

For CDDA, I strongly suggest/recommend using FLAC.
Unless you really have some particular need for uncompressed audio, then (libre-)lossless makes the most sense (using high compression).

Then, for your portable player, if it can't play lossless natively or storage capacity is a constraint, then generate lossy files from the FLACs.

> Astel & Kern SP3000

Based on a quick [Web-index query](//searx.tuxcloud.net/search?q=Astel+%26+Kern+SP3000+formats), your player can indeed play FLACs natively. 👍

> i have decided to just keep them as wav files since there's plenty of storage on the device

If you've got storage to burn, then sure. This may result in lower power-usage (and thus longer runtime on a set of batteries / per-recharge) since there's no processing (in the sense of decoding of a compressed file) for uncompressed audio.
However, your device appears to recharge via USB-C, so you could use an external battery pack (like for charging phones/laptops) if the internal isn't sufficient for a day's listening. I use this approach with multiple devices (nano-computer for [OSM](//www.openstreetmap.org/) surveying (~12 hours straight was my longest single session), and a GoPro (Hero Mk 11, for capturing evidence)), to great effect.

Though, FLAC was designed to be low-demand for decoding (all the complexity is kept on the encoding side, since you encode once but decode many times), and you'd thus be able to fit more audio on the device's storage.

Else, if not FLAC, then … well, that device doesn't seem to be able to play Opus, but it should be able to play Vorbis (often (mistakenly/incorrectly) referred to by the name of the media container format it usually comes in; Ogg).

Apparently, that player can use external SD cards; so effectively unlimited storage. Though, uncompressed seems wasteful (you'll have to buy greater capacity cards, and more of them); since it can play FLAC natively, would seem best to use that; preserves maximal fidelity while being efficient with storage.

Though, the prices I'm noticing, for that player … over $3000?! Really? Yikes.
Could achieve (much) the same result with an Android nano-computer running [VLC](//www.videolan.org/vlc/), an external (USB) DAC [I don't have any pocket-sized ones to recommend] & [headphone-amp](//www.sounddevices.com/product/hx-3/), for significantly less money.
I'm intending to spend less (~2k) on a whole desktop headphone-based chain (including an Allen & Heath (analogue) mixer!), with each component being high-end.

But, if you've got money to burn, too, then go for it. I could make some recommendations for at-home listening, in that case.

> Focal Utopia headphones

Not heard of them, before (and I've worked as an AV tech, before). Interesting. Thanks for the pointer.
